HW1/P2/P2b.py

```diff
@@ -1,3 +1,23 @@
+import numpy as np
 from P2 import *
 
-# Your code here
+import findspark
+findspark.init()
+
+import pyspark
+sc = pyspark.SparkContext()
+
+sc.setLogLevel('WARN')
+
+x = np.random.permutation(2000)
+y = np.random.permutation(2000)
+grid = sc.parallelize(x, 10).cartesian(sc.parallelize(y, 10))
+image = grid.map(lambda (x, y): ((x, y), mandelbrot((x / 500.) - 2, (y / 500.) - 2)))
+draw_image(image)
+
+partition_values = sum_values_for_partitions(image).collect()
+plt.hist(partition_values)
+plt.xlabel('Number of Iterations')
+plt.ylabel('Number of Partitions')
+plt.title('Distribution of Partitions by Number of Iterations')
+plt.savefig('P2b_hist.png')
```

Comment:
It seems like the graph should have less variance in how much work each partition is doing. I used a random partition as well and had much less variance. On a quick glance I can't tell why that is, but it's quite possibly due to the np.random vs partitionby(lambda x: randint) distinction that I brought up earlier